aicsimageio/tests/test_transforms.py

```diff
@@ -80,6 +138,54 @@
             None,
             marks=pytest.mark.raises(exception=ConflictingArgumentsError),
         ),
+        pytest.param(
+            np.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": [0, 1, 7]},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        pytest.param(
+            da.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": [0, 1, 7]},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        pytest.param(
+            np.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": (0, 1, 7)},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        pytest.param(
+            da.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": (0, 1, 7)},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        pytest.param(
+            np.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": range(7)},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
+        pytest.param(
+            da.zeros((6, 200, 400)),
+            "ZYX",
+            "TYXCZX",
+            {"Z": range(7)},
+            None,
+            marks=pytest.mark.raises(exception=IndexError),
+        ),
```

Comment:
might as well add a slice case too?

Reply:
slices don't have index errors. If they can't return anything they result in an empty list.

Reply:
I don't understand - not all of the TYXCZX cases need to expect IndexError, right?

Reply:
Everything except slices imo. How it currently works is that if a slice would result in zero results for that dimension it creates an empty dimension.

```python
print(img.dask_data.shape)  # (1, 1, 6, 70, 233, 345)
print(img.get_image_dask_data("CYX", C=slice(7, 10, 2)).shape)  # (0, 233, 345)
```

Reply:
`slice` objects are normal standard library python so they are special case in my mind but can easily be convinced otherwise